### ISSUE TITLE
[ENH] Add MultiLoss Support and nn.Module Loss Adapters to ptf-v2

### ISSUE THREAD

**Issue by vinitjain2005** [jdb78/pytorch-forecasting#2124]
## Description

Currently, ptf-v2 allows basic loss handling, but there is no standardized support for:

- Combining multiple losses (MultiLoss support)
- Clean integration of arbitrary `torch.nn.Module` losses
- Weighted loss combinations
- A unified adapter abstraction for handling single and multiple loss configurations

As mentioned in the v2 roadmap discussion, adding MultiLoss support and proper loss adapters is important to:

- Achieve feature parity with v1
- Validate the v2 training pipeline design
- Improve flexibility for custom user-defined losses
- Ensure compatibility with future global/foundation model support

---

### 1️⃣ Introduce a `LossAdapter` abstraction

Create a wrapper class that:

- Accepts:
  - A single `nn.Module` loss
  - A list/tuple of losses
  - Optional weights for each loss
- Standardizes output to a consistent format
- Integrates cleanly with v2 `step()` logic

---

### 2️⃣ Add MultiLoss Support

Enable:

- Multiple loss functions combined into a weighted sum
- Proper logging of individual component losses
- Compatibility with existing metric logging

Example usage:

```
```python
loss = MultiLoss(
    losses=[nn.MSELoss(), nn.L1Loss()],
    weights=[0.7, 0.3]
)

```
Github: @vinitjain2005 

@phoeenniixx please assign these task to me so i can start working on these

**Comment by phoeenniixx** [comment]
Hi @vinitjain2005, I think @Nidhicodes is working on `nn` losses adapter (#2073) and @Soham-47 is working on multiloss support (#2122)
